osgar/tools/strip.py

```diff
@@ -0,0 +1,43 @@
+"""
+  Remove some streams from logfile
+"""
+from datetime import timedelta
+
+from osgar.logger import LogReader, LogWriter, lookup_stream_id
+from osgar.lib.serialize import deserialize
+
+
+def strip_logfile(logfile, only_stream, outfile):
+    if outfile is None:
+        outfile = logfile[:-4] + '-strip' + logfile[-4:]
+    print('Out:', outfile)
+    with LogReader(logfile, only_stream_id=only_stream) as log:
+        with LogWriter(filename=outfile, start_time=log.start_time) as out:
+            for timestamp, stream_id, data in log:
+                out.write(stream_id=stream_id, data=data, dt=timestamp)
+
+
+def main():
+    import argparse
+
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument('logfile', help='input log file')
+    parser.add_argument('--stream', help='stream ID or name', nargs='+')
+    parser.add_argument('--out', '-o', help='output logfile')
+    args = parser.parse_args()
+
+    if args.stream is None:
+        only_stream = None
```

Comment:
What is the intent here? What happens when you don't say which streams to keep?

BTW: it is called `strip` but the argument is "streams to keep". I don't have any suggestion here as the name `strip` makes sense and also listing the streams to keep does. :-/

Reply:
By default `None` means "all" so it will be copy 1:1 - no reasonable use case now except check, that the "stripped" result is identical to source.
The short explanation is that I needed that tool and I was not sure what command line interface would suit me (or others), so I planned to leave it to the next iteration. Now I know that there are at least two use cases:

1. lidarview (so you want pose2d, scan, potentially image)
2. replay (so you want all inputs and outputs of given module, "app" in my case)

Reply:
maybe two parameters `--keep` and `--remove`?

Reply:
sounds good :+1: